oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java

```diff
@@ -284,6 +289,11 @@ public void prepare() throws ServiceNotProvidedException, ModuleStartException {
         TopNStreamProcessor.getInstance().setTopNWorkerReportCycle(moduleConfig.getTopNReportPeriod());
         apdexThresholdConfig = new ApdexThresholdConfig(this);
         ApdexMetrics.setDICT(apdexThresholdConfig);
+
+        // TCP manager
+        tcpServerManager = new TCPServerManager(moduleConfig.getTcpHost(),
+            moduleConfig.getTcpBossGroupThreadCount(), moduleConfig.getTcpWorkerGroupThreadCount());
+        this.registerServiceImplementation(TCPBinderRegister.class, new TCPBinderRegisterImpl(tcpServerManager));
```

Comment:
Why TCP server is being injected in the core level? TCP(a.k.a. private network protocol) server for a specific ecosystem only.

Reply:
Here is inject the TCP server manager, all TCP server depends on `TCPBinder`. Declare in here because I need to share the TCP connect and handle thread pool.

Reply:
> because I need to share the TCP connect

I don't agree, what case you could share 2 kinds of TCP protocol with one server?

Reply:
Zabbix server with its TCP protocol is very unique, how this could share with others? Could you provide a real or closing real(in theory) case to support this?

Reply:
Change to define TCP protocol in the Zabbix receive module.